chronos_bin/src/postgres/pg.rs

```diff
@@ -61,26 +61,32 @@ struct PgAccess {
 }
 
 impl PgAccess {
-    pub async fn get_txn(&mut self) -> PgTxn {
+    pub async fn build_txn(&mut self) -> Result<PgTxn, PgError> {
         let txn = self
             .client
             .build_transaction()
             .isolation_level(tokio_postgres::IsolationLevel::RepeatableRead)
-            .start()
-            .await
-            .unwrap();
-        PgTxn { txn }
+            .start();
+
+        match txn.await {
+            Ok(txn) => Ok(PgTxn { txn }),
+            Err(e) => {
+                error!("Unable to start transaction: {}", e);
+                Err(PgError::UnknownException(e))
+            }
+        }
     }
 }
 
 impl Pg {
     pub async fn new(pg_config: PgConfig) -> Result<Self, PgError> {
+        let port = pg_config.port.parse::<u16>().unwrap_or(0); // make the connection fail and send pack PgError
```

Comment:
typo 'pack' > 'back'

Reply:
If port is not configured while it  is required, there is no point retrying. I would consider it as fatal error and fail.